core/src/trezor/ui/layouts/tt_v2/__init__.py

```diff
@@ -1053,3 +1082,50 @@ async def request_pin_on_device(
             raise wire.PinCancelled
         assert isinstance(result, str)
         return result
+
+
+class RustProgress:
+    def __init__(
+        self,
+        title: str,
+        description: str | None = None,
+        indeterminate: bool = False,
+    ):
+        self.description = description or ""
+        self.layout: Any = trezorui2.show_progress(
+            title=title.upper(),
+            indeterminate=indeterminate,
+            description=None if description is None else lambda: self.description,
+        )
+        ui.backlight_fade(ui.style.BACKLIGHT_DIM)
+        ui.display.clear()
+        self.layout.attach_timer_fn(self.set_timer)
```

Comment:
The progress indicators should never set timers because they aren't integrated into the main loop like normal layouts. Using `NotImplementedError` here is confusing though, changed to `RuntimeError` in ea3e9a08fa2219e8f2129c107e4175f81c8c624a.